TheAmazingAudioEngine/AEAudioUnitFileStreamer.m

```diff
@@ -0,0 +1,395 @@
+    if(channelIsPlaying && !super.channelIsPlaying) {
+        if(_playbackDelay != 0) {
+            
+            NSError* error;
+            [self setupPlayRegion:&error];
+            if(error) {
+                NSLog(@"Error setting up audio file region for playback. %@",error.description);
+            }
+        }
+    }
+    
+    // Set the variable
+    super.channelIsPlaying = channelIsPlaying;
+}
+
+// MARK: Render Callbacks
+static OSStatus postRenderCallback(void *							inRefCon,
+                                  AudioUnitRenderActionFlags *	ioActionFlags,
+                                  const AudioTimeStamp *			inTimeStamp,
+                                  UInt32							inBusNumber,
+                                  UInt32							inNumberFrames,
+                                  AudioBufferList *				ioData) {
+    
+    if(*ioActionFlags & kAudioUnitRenderAction_PostRender) {
+        __unsafe_unretained AEAudioUnitFileStreamer* THIS = (__bridge __unsafe_unretained AEAudioUnitFileStreamer*)inRefCon;
+    
+        int32_t playhead = THIS->_playhead;
+        int32_t originalPlayhead = playhead;
+    
+        playhead += inNumberFrames;
+        
+        if(playhead >= THIS->_lengthInFrames + THIS->_playbackDelay) {
+            // If not looping, end the track
+            if( !THIS->_looping ) {
+                
+                // If the playhead is too far, we have to null out the relevant buffer data
+                char *audioPtrs[ioData->mNumberBuffers];
+                for ( int i=0; i<ioData->mNumberBuffers; i++ ) {
+                    audioPtrs[i] = ioData->mBuffers[i].mData;
+                }
+                
+                // Calculate the offset
+                if(THIS->_controllerDescription.mChannelsPerFrame == 1) {
+                    int totalBytes = ioData->mBuffers[0].mDataByteSize;
+                    int frameOffset = inNumberFrames - (playhead - (THIS->_lengthInFrames + THIS->_playbackDelay));
+                    int byteOffset = (frameOffset / inNumberFrames * totalBytes);
+                    int bytesToOverwrite = totalBytes - byteOffset;
+
+                    for ( int i=0; i<ioData->mNumberBuffers; i++ ) {
+                        memset(audioPtrs[i] + byteOffset,0,bytesToOverwrite);
+                    }
+                }
+                else {
+                    int totalBytes = ioData->mBuffers[0].mDataByteSize;
+                    int channelsPerFrame = THIS->_controllerDescription.mChannelsPerFrame;
+                    for ( int i=0; i<ioData->mNumberBuffers; i++ ) {
+                        // TODO: Vectorize this
+                        for(int byteIndex = channelsPerFrame - 1 ; byteIndex < totalBytes ; byteIndex += channelsPerFrame) {
+                            memset(audioPtrs[i] + byteIndex,0,1);
+                        }
+                    }
+                }
+                    
+                THIS.channelIsPlaying = NO;
```

Comment:
Right! This can be fixed simply by changing line 373 to

THIS->channelIsPlaying = NO;